flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetColumnarRowInputFormat.java

```diff
@@ -0,0 +1,189 @@
+
+/**
+ * A {@link ParquetVectorizedInputFormat} to provide {@link RowData} iterator.
+ * Using {@link ColumnarRowData} to provide a row view of column batch.
+ */
+public class ParquetColumnarRowInputFormat extends ParquetVectorizedInputFormat<RowData> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final RowType producedType;
+
+	/**
+	 * Constructor to create parquet format without other fields.
+	 */
+	public ParquetColumnarRowInputFormat(
+			Configuration hadoopConfig,
+			String[] projectedFields,
+			LogicalType[] projectedTypes,
+			int batchSize,
+			boolean isUtcTimestamp,
+			boolean isCaseSensitive) {
+		this(
+				hadoopConfig,
+				projectedFields,
+				projectedTypes,
+				RowType.of(projectedTypes, projectedFields),
+				ColumnBatchFactory.DEFAULT,
+				batchSize,
+				isUtcTimestamp,
+				isCaseSensitive);
+	}
+
+	/**
+	 * Constructor to create parquet format with other fields created by {@link ColumnBatchFactory}.
+	 */
+	public ParquetColumnarRowInputFormat(
+			Configuration hadoopConfig,
+			String[] projectedFields,
+			LogicalType[] projectedTypes,
+			RowType producedType,
+			ColumnBatchFactory batchFactory,
+			int batchSize,
+			boolean isUtcTimestamp,
+			boolean isCaseSensitive) {
+		super(
+				new SerializableConfiguration(hadoopConfig),
+				projectedFields,
+				projectedTypes,
+				batchFactory,
+				batchSize,
+				isUtcTimestamp,
+				isCaseSensitive);
+		this.producedType = producedType;
+	}
+
+	@Override
+	protected ParquetReaderBatch<RowData> createReaderBatch(
+			WritableColumnVector[] writableVectors,
+			VectorizedColumnBatch columnarBatch,
+			Pool.Recycler<ParquetReaderBatch<RowData>> recycler) {
+		return new ColumnarRowReaderBatch(writableVectors, columnarBatch, recycler);
+	}
+
+	@Override
+	public TypeInformation<RowData> getProducedType() {
+		return InternalTypeInfo.of(producedType);
+	}
+
+	private static class ColumnarRowReaderBatch extends ParquetReaderBatch<RowData> {
+
+		private final ColumnarRowIterator result;
+
+		private ColumnarRowReaderBatch(
+				WritableColumnVector[] writableVectors,
+				VectorizedColumnBatch columnarBatch,
+				Pool.Recycler<ParquetReaderBatch<RowData>> recycler) {
+			super(writableVectors, columnarBatch, recycler);
+			this.result = new ColumnarRowIterator(new ColumnarRowData(columnarBatch), this::recycle);
+		}
+
+		@Override
+		public RecordIterator<RowData> convertAndGetIterator(long rowsReturned) {
+			result.set(columnarBatch.getNumRows(), rowsReturned);
+			return result;
+		}
+	}
+
+	/**
+	 * Create a partitioned {@link ParquetColumnarRowInputFormat}, the partition columns can be
+	 * generated by {@link Path}.
+	 */
+	public static ParquetColumnarRowInputFormat createPartitionedFormat(
+			Configuration hadoopConfig,
+			RowType tableType,
+			List<String> partitionKeys,
+			String defaultPartValue,
+			int[] selectedFields,
+			PartitionValueConverter valueConverter,
```

Comment:
This can be improved when it takes a Split in the create/restore method, not just the path and position. Then we can get partition directly.